measure_extinction/extdata.py

```diff
@@ -551,11 +551,32 @@ def trans_elv_alav(self, av=None, akav=0.112):
             if av is None:
                 if "AV" not in self.columns.keys():
                     self.calc_AV(akav=akav)
-                av = _get_column_val(self.columns["AV"])
+            fullav = self.columns["AV"]
+            if len(np.atleast_1d(fullav)) == 1:
+                fullav = np.array([fullav, 0.0])
+            elif len(np.atleast_1d(fullav)) == 3:
+                fullav = np.array([fullav[0], 0.5 * (fullav[1] + fullav[2])])
             for curname in self.exts.keys():
-                self.exts[curname] = (self.exts[curname] / av) + 1
-                self.uncs[curname] /= av
-            # update the extinction curve type
+                # special case for the E(lambda - V) = 0 see below
+                zvals = (self.exts[curname] == 0) & (self.npts[curname] > 0)
+                # formal error propagration where zero extinctions do not
+                # require separate treatment to avoid divide by zero errors
+                self.uncs[curname] = (
+                    np.sqrt(
+                        np.square(self.uncs[curname])
+                        + np.square(self.exts[curname] * fullav[1] / fullav[0])
+                    )
+                    / fullav[0]
+                )
+
+                self.exts[curname] = (self.exts[curname] / fullav[0]) + 1
+                # replace the V band uncertainty with the A(V) uncertainty
+                # as this is the only term nominally in the A(lam)/A(V) extinction
+                # that is by definition 1.
+                #  zvals is defined to only be True for V band
+                if np.sum(zvals) > 0:
+                    self.uncs[curname][zvals] = fullav[1]
```

Comment:
I don't see why this is still needed?

Reply:
Also I don't think this is correct. A(lambda)/A(V) is much smaller than A(V), so using the error on A(V) will be a very big error bar for the V-band.

Reply:
The thought here is that the uncertainty on A(lambda)/A(V) is all about the uncertainty on A(V) as it is A(V)/A(V).  The formal uncertainty on E(V-V) is based on the measurements of the V mags of the two stars.  But this is part of the uncertainty in A(V).  So just following the standard error propagation would double count the V mag uncertainties.  That's my thinking.  What do you think?

Reply:
I agree that the A(V) unc is not correct.  It should be the fractional uncertainty.

Reply:
I am not sure. Even for the V-band you are calculating A(lambda)/A(V) as E(lambda-V)/A(V)+1, so you are using the E(lambda-V) which is 0, but the uncertainty on that is not. Using the standard error propagation will give for the V-band: ```
np.sqrt(np.square(self.uncs[curname])) / fullav[0]
````

Reply:
Also, I noticed that my uncertainties on E(V-V) are all nan for some reason. There might be another problem there,

Reply:
The issue is not that the standard error propagation is wrong, but it leads to a double counting of the V mag uncertianties on the two stars.  As they are both in the E(V-V) value (or should be - not sure about the nans) *and* the calculation of A(V) through the extra step of shifting the E(lam-V) curve up and down by the V mag uncs (e.g., E(V-V) uncertainties).  Hence, we should just use the A(V) uncertainties as they include the uncertainties form the NIR/MIR E(lam-V) values *and* the V mag uncertainties.  Hopefully this makes sense.

Reply:
Ah, actually, the uncertainty on E(V-V) is probably overestimated as you use the uncertainty on the V-band magnitudes twice. Is that what you mean?

Reply:
Actually, A(V)/A(V)= E(V-V)/A(V) +1 = 0+1 = 1, so I would say that nor A(V) nor E(V-V) had anything to do with this calculation, so the uncertainty is 0?

Reply:
I don't think the uncertainty on E(V-V) is incorrect, just that when we divide by A(V), we need to make sure we aren't double counting uncertainties.  This may be an issue throughout the extinction curve given the V mag uncertainties are included in the E(lam-V) uncertainties at all wavelengths.  So, there may be a double counting at all wavelengths.  Not quite sure how to "fix" this, but I would vote for leaving this for a separate issue/PR.  The worst we are doing is over-estimating the uncertainties.

Reply:
The whole converting E(lam-V) to A(lam)/A(V) (or E(lam-V)/E(B-V)) while getting the uncertainties correct is something that is surprisingly complex.  May need to do some more serious thinking/equations/Monte Carlo simulations to really understand how to do this accurately.

Reply:
But I do think we are doing ok and over-estimating the uncertainties only.  And I think by not too much.  But time will tell.

Reply:
Yes, we should move forward with these changes, and maybe investigate in more detail at a later time.